holidays/countries/nepal.py

```diff
@@ -0,0 +1,441 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.calendars import _CustomIslamicHolidays, _CustomHinduHolidays
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class Nepal(
+    HolidayBase,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+):
+    """Nepal holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Nepal>
+        * <https://narayanilawfirm.org.np/list-of-public-holidays-in-nepal-2079/>
+        * <https://www.timeanddate.com/holidays/nepal/>
+        * <https://kathmandupost.com/national/2018/03/29/government-trims-public-holidays-by-22-days>
+        * <https://kathmandupost.com/national/2021/02/13/two-years-after-shortening-public-holiday-list-government-starts-adding-to-it-again>
+        * <https://english.hamropatro.com/nepali-public-holidays>
+        * <https://english.hamropatro.com/calendar>
+        * <https://www.ashesh.com.np/nepali-calendar/>
+        * <https://www.bolpatra.gov.np/egp/openDateConverter>
+        * <https://www.moha.gov.np/en/page/holidays>
+    """
+
+    country = "NP"
+    # %s (estimated).
+    estimated_label = "%s (estimated)"
+    start_year = 2010
+    supported_categories = (PUBLIC, WORKDAY)
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=NepalHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=NepalIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        StaticHolidays.__init__(self, cls=NepalStaticHolidays)
+        super().__init__(*args, **kwargs)
+
+    def _add_non_continuous_holidays(self, is_workday: bool = False):
+        """Holidays removed by MoHA between 2019-2020."""
+        if (2019 <= self._year <= 2020) == is_workday:
```

Comment:
// no action required: this could be `is` instead of `==` but I know we have another case in `UA` holidays and it's also arguable from the code readability perspective 🤷‍♂️